GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m

```diff
@@ -329,55 +350,57 @@ - (void)handleURLResponse:(nullable NSURLResponse *)response
                    target:(GDTCORTarget)target
                   storage:(id<GDTCORStorageProtocol>)storage
                   batchID:(NSNumber *)batchID {
-  GDTCORLogDebug(@"%@", @"CCT: request completed");
-  if (error) {
-    GDTCORLogWarning(GDTCORMCWUploadFailed, @"There was an error uploading events: %@", error);
-  }
-  NSError *decodingError;
-  GDTCORClock *futureUploadTime;
-  if (data) {
-    gdt_cct_LogResponse logResponse = GDTCCTDecodeLogResponse(data, &decodingError);
-    if (!decodingError && logResponse.has_next_request_wait_millis) {
-      GDTCORLogDebug(@"CCT: The backend responded asking to not upload for %lld millis from now.",
-                     logResponse.next_request_wait_millis);
-      futureUploadTime =
-          [GDTCORClock clockSnapshotInTheFuture:logResponse.next_request_wait_millis];
-    } else if (decodingError) {
-      GDTCORLogDebug(@"There was a response decoding error: %@", decodingError);
+    GDTCORLogDebug(@"%@", @"CCT: request completed");
+    if (error) {
+      GDTCORLogWarning(GDTCORMCWUploadFailed, @"There was an error uploading events: %@", error);
+    }
+    NSError *decodingError;
+    GDTCORClock *futureUploadTime;
+    if (data) {
+      gdt_cct_LogResponse logResponse = GDTCCTDecodeLogResponse(data, &decodingError);
+      if (!decodingError && logResponse.has_next_request_wait_millis) {
+        GDTCORLogDebug(@"CCT: The backend responded asking to not upload for %lld millis from now.",
+                       logResponse.next_request_wait_millis);
+        futureUploadTime =
+            [GDTCORClock clockSnapshotInTheFuture:logResponse.next_request_wait_millis];
+      } else if (decodingError) {
+        GDTCORLogDebug(@"There was a response decoding error: %@", decodingError);
+      }
+      pb_release(gdt_cct_LogResponse_fields, &logResponse);
+    }
+    if (!futureUploadTime) {
+      GDTCORLogDebug(@"%@", @"CCT: The backend response failed to parse, so the next request "
+                            @"won't occur until 15 minutes from now");
+      // 15 minutes from now.
+      futureUploadTime = [GDTCORClock clockSnapshotInTheFuture:15 * 60 * 1000];
+    }
+    switch (target) {
+      case kGDTCORTargetCCT:
+        self->_CCTNextUploadTime = futureUploadTime;
+        break;
+
+      case kGDTCORTargetFLL:
+        // Falls through.
+      case kGDTCORTargetINT:
```

Comment:
I'm not sure we should apply the next upload timeout set for another backend. is `kGDTCORTargetINT` a Firelog endpoint as well?

Reply:
Yes - this is for crashlytics but inside of google3.